Common/Tests/Utilities/FileUtils.cs

```diff
@@ -167,6 +170,14 @@ public static class FileUtils {
             foreach (var newFile in newFiles) {
                 var copyFrom = Path.Combine(sourceDir, newFile);
                 var copyTo = Path.Combine(destDir, newFile);
+
+                if (tryHardLinkFirst) {
```

Comment:
This is mainly used for "copying" binary dependencies for tests into a directory that has about 50% of them already. There are very few subdirectories, so very little benefit there.

Also, once you create a symlink or a junction to a directory (there's no such thing as a directory hardlink to my knowledge) it gets really hard to delete things, whereas file hardlinks just delete normally and remove the reference. Plus we really don't want to modify the originals (I started with setting the read-only bit here, but that applies to all links to the same file and so it broke subsequent builds).